src/basic/fileio.c

```diff
@@ -799,7 +795,7 @@ int read_line(FILE *f, size_t limit, char **ret) {
                                 /* Previous char was a NUL? This is not an EOL, but the previous char was? This type of
                                  * EOL marker has been seen right before? In either of these three cases we are
                                  * done. But first, let's put this character back in the queue. */
-                                assert_se(ungetc(c, f) != EOF);
+                                assert_se(ungetc((unsigned char) c, f) != EOF);
```

Comment:
I was quite surprised by the cast here. But it seems `ungetc` doesn't care if it gets a positive or a negative value. So is this is not incorrect, but I'd prefer to have the cast dropped, unless there's some good reason to keep it.